nx/lib/nx.ex

```diff
@@ -6658,6 +6658,217 @@ defmodule Nx do
     impl!(tensor).cholesky(out, tensor)
   end
 
+  @doc """
+  Calculates the p-norm of a tensor.
+
+  For the 0-norm, the norm is the number of non-zero elements in the tensor.
+
+  ### Examples
+
+    iex> Nx.norm(Nx.tensor([3, 4]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: 1)
+    #Nx.Tensor<
+      f64
+      7.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      4
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      3
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4, 0, 0]), ord: 0)
+    #Nx.Tensor<
+      u64
+      2
+    >
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :fro)
+    ** (RuntimeError) expected a 2-D tensor. Got a 1-D tensor.
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :nuc)
+    ** (RuntimeError) expected a 2-D tensor. Got a 1-D tensor.
+
+    iex> Nx.norm(Nx.tensor([[3, -1], [2, -4]]), ord: -1)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: 1)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]), ord: :fro)
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 4], [0, -4]]), axes: [1], keep_axes: true)
+    #Nx.Tensor<
+      f64[2][1]
+      [
+        [5.0],
+        [4.0]
+      ]
+    >
+
+  ### Caveats
+
+  For big values of p, f64 rounding errors come into play
+  """
+
+  def norm(t_raw, input_opts \\ []) when is_list(input_opts) do
+    t = tensor!(t_raw)
+
+    case shape(t) do
+      {_, _} ->
+        :ok
+
+      {_} ->
+        :ok
+
+      s ->
+        raise "expected 1-D or 2-D tensor. Received a tensor with shape #{inspect(s)} instead."
+    end
+
+    rank = rank(t)
+    ord = get_norm_ord!(input_opts, rank)
+
+    default_axes_opts = [axes: nil, keep_axes: false]
+
+    axes_opts = Keyword.merge(default_axes_opts, Keyword.take(input_opts, [:axes, :keep_axes]))
```

Comment:
Also, those are default values, so this should be fine:

```suggestion
    axes_opts = Keyword.take(input_opts, [:axes, :keep_axes])
```